sourcerer-test-utils/src/main/java/org/elder/sourcerer/eventstore/test/rules/EventStoreRule.java

```diff
@@ -0,0 +1,75 @@
+package org.elder.sourcerer.eventstore.test.rules;
+
+import com.mashape.unirest.http.Unirest;
+import com.mashape.unirest.http.exceptions.UnirestException;
+import com.shazam.tocker.AliveStrategies;
+import com.shazam.tocker.DockerInstance;
+import com.shazam.tocker.PortMap;
+import com.shazam.tocker.RunningDockerInstance;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * JUnit rule for starting and stopping an EventStore instance using docker.
+ */
+public class EventStoreRule extends ExternalResource {
+    private static final String SERVICE_HOSTNAME = "localhost";
+    private static final int SERVICE_PORT = 1113;
+    private static final int PROBE_PORT = 2113;
+    private static final Logger logger = LoggerFactory.getLogger(EventStoreRule.class);
+    private final DockerInstance instance = DockerInstance
+            .fromImage("eventstore/eventstore:release-4.0.1")
+            .withContainerName("eventstore-test-container")
+            .mappingPorts(
+                    PortMap.ephemeral(SERVICE_PORT),
+                    PortMap.ephemeral(PROBE_PORT))
+            .build();
+    private RunningDockerInstance store;
+    private List<Runnable> cleanup = new ArrayList<>();
+
+    @Override
+    protected void before() throws Throwable {
+        logger.info("Starting EventStore");
+        store = instance.run(AliveStrategies.retrying(livenessProbe(), 100, 100));
+    }
+
+    @Override
+    protected void after() {
+        for (Runnable job : cleanup) {
+            job.run();
+        }
+        instance.stop();
+    }
+
+    public EventStoreRule start() {
+        return this;
+    }
+
+    private Function<RunningDockerInstance, Boolean> livenessProbe() {
+        return instance -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Rename "instance" which hides the field declared at line 25. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.elder.org/coding_rules#rule_key=squid%3AHiddenFieldCheck)